drivers/UnbufferedSerial.h

```diff
@@ -0,0 +1,162 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_UNBUFFERED_SERIAL_H
+#define MBED_UNBUFFERED_SERIAL_H
+
+#include "platform/platform.h"
+
+#if DEVICE_SERIAL || defined(DOXYGEN_ONLY)
+
+#include "drivers/SerialBase.h"
+#include "platform/FileHandle.h"
+#include "platform/mbed_toolchain.h"
+#include "platform/NonCopyable.h"
+#include <cstdarg>
+
+namespace mbed {
+
+/**
+ * \defgroup drivers_UnbufferedSerial UnbufferedSerial class
+ * \ingroup drivers-public-api-uart
+ * @{
+ */
+
+/**
+ * Class implementation for unbuffered I/O for an interrupt driven application
+ * or one that needs to have more control.
+ */
+class UnbufferedSerial:
+    public SerialBase,
+    public FileHandle,
+    private NonCopyable<UnbufferedSerial>
+{
+public:
+    /**
+     * Create a serial port instance connected to the specified transmit and
+     * receive pins, with the specified baud rate.
+     *
+     *  @param tx Transmit pin
+     *  @param rx Receive pin
+     *  @param baud The baud rate of the serial port (optional, defaults to MBED_CONF_PLATFORM_DEFAULT_SERIAL_BAUD_RATE)
+     *
+     *  @note
+     *    Either tx or rx may be specified as NC if unused
+     */
+    UnbufferedSerial(
+        PinName tx,
+        PinName rx,
+        int baud = MBED_CONF_PLATFORM_DEFAULT_SERIAL_BAUD_RATE
+    );
+
+    /** Write the contents of a buffer to a file
+     *
+     *  Follows POSIX semantics:
+     *
+     * * if blocking, block until all data is written
+     * * if no data can be written, and non-blocking set, return -EAGAIN
+     * * if some data can be written, and non-blocking set, write partial
```

Comment:
You should say what this function does in the case of UnbufferedSerial i.e. blocking.

Reply:
Please review [this](https://github.com/hugueskamba/mbed-os/pull/5/commits/ee5808441ce815e9d6061c19622c2ef8b5fe972d) and resolve if satisfactory.